pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,313 @@
+            Design Using a Minimal-Phase Target Function from Regularization,”
+            Convention Paper 6929 Presented at the 121st Convention, 2006
+            October 5–8, San Francisco, CA, USA.
+    """ # noqa: E501
+
+    def __init__(self, regu_type: str="") -> None:
+
+        self._regu_type = regu_type
+
+        # throw error if object is instanced without classmethod
+        if regu_type == "":
+            raise RuntimeError("Regularization objects must be created "
+                             "using a classmethod.")
+
+    def __repr__(self):
+        return f"Regularization object with regularization "\
+            f"of type '{self._regu_type}'."
+
+    @classmethod
+    def from_frequency_range(cls, frequency_range):
+        r"""
+        Regularization from frequency range.
+        Defines a frequency range within which the regularization factor is set
+        to ``0``. Outside the frequency range the regularization factor is
+        ``1`` and can be scaled using the ``beta`` parameter.
+        The regularization factors are cross-faded using a raised cosine
+        window function with a width of :math:`\sqrt{2}f` above and below the
+        given frequency range.
+
+        Parameters
+        ----------
+        frequency_range : array, tuple
+            Tuple with two values for the lower and upper frequency limit.
+        """
+        instance = cls(regu_type = "frequency range")
+
+        if len(frequency_range) != 2:
+            raise ValueError(
+                "The frequency range needs to specify lower and upper limits.")
+
+        instance._frequency_range = np.asarray(frequency_range)
+
+        return instance
+
+    @classmethod
+    def from_magnitude_spectrum(cls, regularization):
+        """
+        Regularization from magnitude spectrum.
+        Regularization factors passed as :py:class:`~pyfar.Signal`. The factors
+        have to match the number of bins of the signal to be inverted.
+
+        Parameters
+        ----------
+        regularization : pf.Signal
+            Regularization as pyfar Signal.
+        """
+        if not isinstance(regularization, pf.Signal):
+            raise ValueError(
+                "Regularization must be a pyfar.Signal object.")
+
+        instance = cls(regu_type = "magnitude spectrum")
+        instance._regu = regularization
+
+        return instance
+
+    def invert(self, signal, beta=1, target=None,
+               normalize_regularization=None):
+        r"""Invert the spectrum of a signal applying frequency dependent
+        regularization.
+
+        Parameters
+        ----------
+        signal : pf.Signal
+            Signal to be inverted.
+        beta : float, optional
+            Beta parameter to control the amount of regularization.
+        target : pf.Signal, optional
+            Target function for the regularization.
+        normalize_regularization : str
```

Comment:
```suggestion
        normalize_regularization : str, optinal
```

and we need to add, by defauld no normalization